### DISCUSSION TITLE
Autofix workflow on a pull request invalidates previous check state without triggering new checks

### DISCUSSION THREAD

**Discussion by jaredly** [community/community#25874] [Actions]
<p>I’ve got a workflow that runs on pull requests, to autofix formatting issues with <code>prettier</code>, and then commits + pushes the fixes (if any) back up to the pull-requesting branch.</p>
<p>Essentially:</p>
<pre><code>yarn prettier

git commit -am "Committing autofixes"

ORIGIN=https://${GITHUB_ACTION}:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git

echo "Pushing to repo..."
git fetch $ORIGIN ${GITHUB_HEAD_REF}
git merge FETCH_HEAD -m "Merging in remote"
git push $ORIGIN HEAD:${GITHUB_HEAD_REF}
</code></pre>
<p>This works great, but the resulting “Committing autofixes” commit doesn’t appear to trigger any checks. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/bc50e01eab7d56eea63e31705e0af68b2d6a9115.png" data-download-href="/uploads/short-url/qRV2rcJKJbMcY2vvzMGqHM2z4tT.png?dl=1" title="Screen Shot 2019-10-04 at 2.07.18 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/112170/181093793-33455c2b-e73a-47ce-985e-88f46b0c8bbd.png" alt="Screen Shot 2019-10-04 at 2.07.18 PM.png" data-base62-sha1="qRV2rcJKJbMcY2vvzMGqHM2z4tT" width="690" height="113" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/bc50e01eab7d56eea63e31705e0af68b2d6a9115_2_690x113.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/bc50e01eab7d56eea63e31705e0af68b2d6a9115_2_1035x169.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/b/bc50e01eab7d56eea63e31705e0af68b2d6a9115_2_1380x226.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/bc50e01eab7d56eea63e31705e0af68b2d6a9115_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-10-04 at 2.07.18 PM.png</span><span class="informations">1410×232 30.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I guess this makes sense (to prevent actions that trigger themselves in an infinite loop), but the problem is that the pull request no longer thinks it has any checks – even if the previous checks had failed.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0bc5d1f5446915d112b2819d342096c9f67ce75c.png" data-download-href="/uploads/short-url/1G94X5LzmJA3csWOiDXrFKULynW.png?dl=1" title="Screen Shot 2019-10-04 at 2.15.01 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/112170/181093794-464e4a4a-7252-4221-b50e-6747c0265cb8.png" alt="Screen Shot 2019-10-04 at 2.15.01 PM.png" data-base62-sha1="1G94X5LzmJA3csWOiDXrFKULynW" width="690" height="177" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/0bc5d1f5446915d112b2819d342096c9f67ce75c_2_690x177.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/0bc5d1f5446915d112b2819d342096c9f67ce75c_2_1035x265.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/0bc5d1f5446915d112b2819d342096c9f67ce75c_2_1380x354.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0bc5d1f5446915d112b2819d342096c9f67ce75c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-10-04 at 2.15.01 PM.png</span><span class="informations">1534×394 43 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’m “working around” this problem by, after autofixing happens, making an empty commit locally and pushing it up, e.g. git commit --allow-empty -m bump; git push, which triggers the checks again and brings things back to normal.</p>
<p>And then the pull-request looks like this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/62621e003ed575e4be145929b856c0c2d4208b24.png" data-download-href="/uploads/short-url/e2l0rJX63BSheDitfqPthKty3hW.png?dl=1" title="Screen Shot 2019-10-04 at 2.08.51 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/112170/181093795-9d851408-8455-4b36-8e8d-7e4968785851.png" alt="Screen Shot 2019-10-04 at 2.08.51 PM.png" data-base62-sha1="e2l0rJX63BSheDitfqPthKty3hW" width="690" height="394" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/62621e003ed575e4be145929b856c0c2d4208b24_2_690x394.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/62621e003ed575e4be145929b856c0c2d4208b24_2_1035x591.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/6/62621e003ed575e4be145929b856c0c2d4208b24_2_1380x788.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/62621e003ed575e4be145929b856c0c2d4208b24_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-10-04 at 2.08.51 PM.png</span><span class="informations">1538×880 96 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Has anyone else run into this before? Is there a better workflow for having an action that commits autofixes back to a pull request?</p>

**Comment by jaredly** [comment]
<p>I also tried closing &amp; reopening the pull request after the autofix (which the docs claim will trigger a pull_request workflow), but it didn’t :confused:</p>

**Comment by wipe2238** [comment, marked as answer]
<p>Provided GITHUB_TOKEN is not allowed to trigger checks, you need to generate and use own/bot’s account/any other token and it will work just fine.</p>
<p>If using bot account to do formatting, you probably don’t want to waste time on formatting just-formatted code…</p>
<pre><code>on: push
jobs:
 format-source:
  runs-on: ubuntu-latest
  if: github.actor != 'my_bot_account'
</code></pre>
<p>…and works great to prevent infinite loop you mentioned.</p>

**Reply by ZulluBalti** [reply]
what is 'my_bot_account', should I just put that there?

**Reply by wipe2238** [reply]
@ZulluBalti
It's just username of separate account, which does everything via scripts, actions, etc. using its access token (see 'tip' in [types of GitHub accounts docs](https://docs.github.com/en/get-started/learning-about-github/types-of-github-accounts#personal-accounts)).
